### DISCUSSION TITLE
Tools/DSM FWDPRG_05_Color

### DISCUSSION THREAD

**Discussion by KevinP3C** [pascallanger/DIY-Multiprotocol-TX-Module#759] [General]
Received error message unexpected symbol near '<' when trying to use forward programing on tx16s with edgetx. Script syntax error banner displayed.

**Comment by frankiearzu** [comment]
Kevin
Make sure that  the subdirectory  DSMLIB is there.  What RX are you using?
I test it on AR631 and others has tested on AR637s, and the Heli FC62xxxxx. 
There is a hack for some RXs, without the hack, it was not loading some menus. It fix a problem on  AR631/AR637x to make the initial setup (after factory reset) to work.   If you are testing with a different RX and you get "Cant read menu data...", maybe that receiver needs the hack too.

your deployment should look:

Deployment:
/SCRIPTS/TOOLS/DsmFwdPrg_05_BW.lua		-- OpenTX/EdgeTx version, no touch (black/white radios)
/SCRIPTS/TOOLS/DsmFwdPrg_05_Color.lua		-- EdgeTX/OpenTx Color version, +touch
/SCRIPTS/TOOLS/DSMLIB/				-- Libraries (ALL CAPITALS) to hide them from the Tools menu
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgLib.lua		-- DSM Protocol Message and Menu engine
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgSIMLib.lua	-- Simulation of AR631 for GUI Development in Companion
/SCRIPTS/TOOLS/DSMLIB/img			-- Images for RX orientations

Looks like pascal already merge my changes to master.. give it another try.

Discusion:
https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/issues/751

**Comment by KevinP3C** [comment]
Thanks for your help. I deleted the script. Downloaded the file again, reinstalled it on the TX16S Radio Master transmitter running Edge TX version 2.7 and had the same warning. Do I have to download multiple scripts to make the program work?Thanks, I am new at this and probably made a mistake somewhere. Thanks againSent from my iPhoneOn Nov 20, 2022, at 11:38 AM, Frankie Arzu ***@***.***> wrote:﻿
Kevin
Make that you have the subdirectory  DSMLIB is there.  What RX are you using?
I test it on AR631 and others has tested on AR637s, and the Heli FC62xxxxx.
your deployment should look:
Deployment:
/SCRIPTS/TOOLS/DsmFwdPrg_05_BaW.lua		-- OpenTX/EdgeTx version, no touch (black/white radios)
/SCRIPTS/TOOLS/DsmFwdPrg_05_Color.lua		-- EdgeTX/OpenTx Color version, +touch
/SCRIPTS/TOOLS/DSMLIB/				-- Libraries (ALL CAPITALS) to hide them from the Tools menu
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgLib.lub		-- DSM Protocol Message and Menu engine
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgSIMLib.lub	-- Simulation of AR631 for GUI Development in Companion
/SCRIPTS/TOOLS/DSMLIB/img			-- Images for RX orientations
My very last pull-request has not been merged into master, but the old one should work (without orientation images).
Discusion:
#751
Latest code:
https://github.com/frankiearzu/DIY-Multiprotocol-TX-Module/tree/frankie-dsm-fwrd-prg-enhancements

**Comment by frankiearzu** [comment]
Kevin,
How are you downloading the files?? do you have GIT setup?? if not using GIT, download a ZIP file with the entire files:
Code->Local->Download ZIP

![Capture](https://user-images.githubusercontent.com/32604366/203177501-b6759eaf-a6e6-4051-acc0-c3ca54687c9b.JPG)

Yes, there are mutiple files. Besides the main script in /SCRIPTS/TOOLS, you have to copy the files inside DSMLIB.

A friend just downloaded and copy it (all 27 files) and works for him on a AR637.    At minimun,  you need the LUA files inside the DSMLIB (2 files) and the main script(Color or BW version, 1 or 2 files).  Images are only used in the RX orientation page, so you can skip them.
The main script will give you an error if it does not find the extra LUA scripts inside the DSMLIB, It will tell you at start that the files inside DSMLIB are missing very explicit.  Have test it on TX16S EdgeTx version 2.7.1 and 2.8 RC4.

Your SD card structure and files should look like this:
/SCRIPTS/TOOLS/DsmFwdPrg_05_BW.lua -- OpenTX/EdgeTx version, no touch (black/white radios)
/SCRIPTS/TOOLS/DsmFwdPrg_05_Color.lua -- EdgeTX/OpenTx Color version, +touch
/SCRIPTS/TOOLS/DSMLIB/ -- Libraries (ALL CAPITALS) to hide them from the Tools menu
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgLib.lua -- DSM Protocol Message and Menu engine
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgSIMLib.lua -- Simulation of AR631 for GUI Development in Companion
/SCRIPTS/TOOLS/DSMLIB/img -- Images for RX orientations  (23 files)


Thank you,
Frankie

**Comment by KevinP3C** [comment]
Thanks I see the files in the file finder program, how do I download them? The receivers I’m dealing with are AR636A’s. Sorry to be a pain Sent from my iPhoneOn Nov 21, 2022, at 5:57 PM, Frankie Arzu ***@***.***> wrote:﻿
Kevin,
Yes, there are mutiple files. Besides the main script in /SCRIPTS/TOOLS, you have to copy the files inside DSMLIB.
A friend just downloaded it (all 27 files) and works for him on a AR637.    At minimun,  you need the LUA files inside the DSMLIB (2 files) and the main script(Color or BW version, 1 or 2 files).  Images are only used in the RX orientation page, so you can skip them.
The main script will give you an error if it does not find the extra LUA scripts inside the DSMLIB, It will tell you at start that the files inside DSMLIB are missing very explicit.  Have test it on TX16S EdgeTx version 2.7.1 and 2.8 RC4.
Your SD card structure and files should look like this:
/SCRIPTS/TOOLS/DsmFwdPrg_05_BW.lua -- OpenTX/EdgeTx version, no touch (black/white radios)
/SCRIPTS/TOOLS/DsmFwdPrg_05_Color.lua -- EdgeTX/OpenTx Color version, +touch
/SCRIPTS/TOOLS/DSMLIB/ -- Libraries (ALL CAPITALS) to hide them from the Tools menu
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgLib.lua -- DSM Protocol Message and Menu engine
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgSIMLib.lua -- Simulation of AR631 for GUI Development in Companion
/SCRIPTS/TOOLS/DSMLIB/img -- Images for RX orientations  (23 files)
Thank you,
Frankie

**Comment by frankiearzu** [comment]
AR636 don’t have forward programming.  I have one an no luck.I think only the helicopter firmware for it has something (blade helps), but haven’t try it yet.If you want to download the files, the best is to download the zip file from GitHub for the project.. sent you a screenshot in my previous message.Frankie On Nov 22, 2022, at 9:43 AM, KevinP3C ***@***.***> wrote:﻿
Thanks I see the files in the file finder program, how do I download them? The receivers I’m dealing with are AR636A’s. Sorry to be a pain Sent from my iPhoneOn Nov 21, 2022, at 5:57 PM, Frankie Arzu ***@***.***> wrote:﻿
Kevin,
Yes, there are mutiple files. Besides the main script in /SCRIPTS/TOOLS, you have to copy the files inside DSMLIB.
A friend just downloaded it (all 27 files) and works for him on a AR637.    At minimun,  you need the LUA files inside the DSMLIB (2 files) and the main script(Color or BW version, 1 or 2 files).  Images are only used in the RX orientation page, so you can skip them.
The main script will give you an error if it does not find the extra LUA scripts inside the DSMLIB, It will tell you at start that the files inside DSMLIB are missing very explicit.  Have test it on TX16S EdgeTx version 2.7.1 and 2.8 RC4.
Your SD card structure and files should look like this:
/SCRIPTS/TOOLS/DsmFwdPrg_05_BW.lua -- OpenTX/EdgeTx version, no touch (black/white radios)
/SCRIPTS/TOOLS/DsmFwdPrg_05_Color.lua -- EdgeTX/OpenTx Color version, +touch
/SCRIPTS/TOOLS/DSMLIB/ -- Libraries (ALL CAPITALS) to hide them from the Tools menu
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgLib.lua -- DSM Protocol Message and Menu engine
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgSIMLib.lua -- Simulation of AR631 for GUI Development in Companion
/SCRIPTS/TOOLS/DSMLIB/img -- Images for RX orientations  (23 files)
Thank you,
Frankie

**Comment by KevinP3C** [comment]
Thanks for your help, Kevin Sent from my iPhoneOn Nov 22, 2022, at 10:51 AM, Frankie Arzu ***@***.***> wrote:﻿
AR636 don’t have forward programming.  I have one an no luck.I think only the helicopter firmware for it has something (blade helps), but haven’t try it yet.If you want to download the files, the best is to download the zip file from GitHub for the project.. sent you a screenshot in my previous message.Frankie On Nov 22, 2022, at 9:43 AM, KevinP3C ***@***.***> wrote:﻿
Thanks I see the files in the file finder program, how do I download them? The receivers I’m dealing with are AR636A’s. Sorry to be a pain Sent from my iPhoneOn Nov 21, 2022, at 5:57 PM, Frankie Arzu ***@***.***> wrote:﻿
Kevin,
Yes, there are mutiple files. Besides the main script in /SCRIPTS/TOOLS, you have to copy the files inside DSMLIB.
A friend just downloaded it (all 27 files) and works for him on a AR637.    At minimun,  you need the LUA files inside the DSMLIB (2 files) and the main script(Color or BW version, 1 or 2 files).  Images are only used in the RX orientation page, so you can skip them.
The main script will give you an error if it does not find the extra LUA scripts inside the DSMLIB, It will tell you at start that the files inside DSMLIB are missing very explicit.  Have test it on TX16S EdgeTx version 2.7.1 and 2.8 RC4.
Your SD card structure and files should look like this:
/SCRIPTS/TOOLS/DsmFwdPrg_05_BW.lua -- OpenTX/EdgeTx version, no touch (black/white radios)
/SCRIPTS/TOOLS/DsmFwdPrg_05_Color.lua -- EdgeTX/OpenTx Color version, +touch
/SCRIPTS/TOOLS/DSMLIB/ -- Libraries (ALL CAPITALS) to hide them from the Tools menu
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgLib.lua -- DSM Protocol Message and Menu engine
/SCRIPTS/TOOLS/DSMLIB/DsmFwPrgSIMLib.lua -- Simulation of AR631 for GUI Development in Companion
/SCRIPTS/TOOLS/DSMLIB/img -- Images for RX orientations  (23 files)
Thank you,
Frankie
